common/performance.py

```
@@ -0,0 +1,721 @@
+from common.metrics import metrics
+
+# ===================================================================
+# Template View Caching (unchanged)
+# ===================================================================
+
+
+class CachedTemplateView(TemplateView):
+    """Template view with automatic caching enabled."""
+
+    @classmethod
+    def as_view(cls, **initkwargs) -> Callable[..., HttpResponse]:
```

Ruff (0.12.2)

Missing type annotation for `**initkwargs`

(ANN003)